cilium-health/launch/endpoint.go

```diff
@@ -224,7 +224,7 @@ func CleanupEndpoint() {
 //
 // CleanupEndpoint() must be called before calling LaunchAsEndpoint() to ensure
 // cleanup of prior cilium-health endpoint instances.
-func LaunchAsEndpoint(baseCtx context.Context, owner regeneration.Owner, n *node.Node, mtuConfig mtu.Configuration) (*Client, error) {
+func LaunchAsEndpoint(baseCtx context.Context, owner regeneration.Owner, n *node.Node, mtuConfig mtu.Configuration, epMgr *endpointmanager.EndpointManager) (*Client, error) {
```

Comment:
This does not need an endpoitmanager, it needs a interface that implements `AddEndpoint`

Reply:
@aanm agreed. Will follow-up with commits that allow us to inject the dependency.